UofT-Frosh-Orientation/orientation-website
# 611 set up nginx on server

## **Contribution:**


### **Issue:**

Closes #611 

### **Accomplishments:**

- Set up Nginx on Beta server
- Wrote documentation on Hosting & Deployment

### **Visuals:** 

![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/43909693/a48aaa8c-1e9f-43b4-8b6a-e4b8b5fe43d9)


## **Details:**

### **Expected Behavior:**

- The full project is up and running on the Beta server and fully accessible.
- You should be able to create an account.

### **Testing Steps:**

Steps required to get expected behavior.

- Go to https://beta.orientation.skule.ca and see that the frontend and backend are fully function
- Create an account

### **Complications:**

- Nginx configuration was very tedious 

### **Resources:**

Any additional resources used outside this in issue.

- Shoutouts last years Tech Team Leads, they were very helpful.
- The DigitalOcean documentation was also very useful.
> - [Initial Server Setup](https://www.digitalocean.com/community/tutorials/initial-server-setup-with-ubuntu-20-04#if-the-root-account-uses-ssh-key-authentication)

> - [How To Install Nginx](https://www.digitalocean.com/community/tutorials/how-to-install-nginx-on-ubuntu-20-04#step-5-setting-up-server-blocks-recommended)

> - [How To Secure Nginx with Let's Encrypt](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04)
